### PR TITLE
Enforce singleton in release only

### PR DESCRIPTION
**Pull request by gnunicorn**
superseeds #2393  .

This allows Windows and Linux developers to run the dev-version _next to_ an installed Acter with its own secret-store identity and not being blocked by their single-window problem.

The core problem on linux is the ``SingleMainWindow=true` flag in the `.desktop`-file. If acter is installed and an  app starts with that same identity, the desktop will switch to the first. This patch now overwrites the identity with a `-dev`-suffixed one to get around the issue, as long as this is a debug build. Thus the desktop environment understoods them as different apps and allows them to be run simultaneously.

### CODE REVIEW

File: app/windows/runner/main.cpp [hunk 1]
```diff
@@ -8,8 +8,11 @@
 
 int APIENTRY wWinMain(_In_ HINSTANCE instance, _In_opt_ HINSTANCE prev,
                       _In_ wchar_t *command_line, _In_ int show_command) {
-
+#ifdef _DEBUG
+  HWND hwnd = ::FindWindow(L"FLUTTER_RUNNER_WIN32_WINDOW", L"Acter-dev");
```
Comment (bitfriend) [inline app/windows/runner/main.cpp:12]: In windows, which stage in compilation will deal with `Acter-dev`?

Comment (gnunicorn) [reply app/windows/runner/main.cpp:12]: To make sure we use a different window system and don't block the installed version.

Comment (bitfriend) [reply app/windows/runner/main.cpp:12]: Then windows and linux will detect release/debug flag using the different logic?

Comment (gnunicorn) [reply app/windows/runner/main.cpp:12]: this is the windows one, yes.

**Pull request merged**